applications/StructuralMechanicsApplication/python_scripts/adaptative_remeshing_structural_mechanics_static_solver.py

```diff
@@ -55,10 +52,19 @@ def get_remeshing_process(self):
         return self._remeshing_process
 
     def _create_remeshing_process(self):
-        if (self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE] == 2):
+        if self.main_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE] == 2:
             remeshing_process = MeshingApplication.MmgProcess2D(self.main_model_part, self.adaptative_remesh_parameters["remeshing_parameters"])
         else:
-            remeshing_process = MeshingApplication.MmgProcess3D(self.main_model_part, self.adaptative_remesh_parameters["remeshing_parameters"])
+            is_surface = False
+            for elem in self.main_model_part.Elements:
+                geom = elem.GetGeometry()
+                if geom.WorkingSpaceDimension() != geom.LocalSpaceDimension():
+                    is_surface = True
+                break
+            if is_surface:
+                remeshing_process = MeshingApplication.MmgProcess3DSurfaces(self.main_model_part, self.adaptative_remesh_parameters["remeshing_parameters"])
+            else:
+                remeshing_process = MeshingApplication.MmgProcess3D(self.main_model_part, self.adaptative_remesh_parameters["remeshing_parameters"])
```

Comment:
can you please make a common-class like you did in contact to not copy-paste code between the solvers?

Reply:
I can, but not today